### ISSUE TITLE
EMF dependencies that aren't available on maven central 

### ISSUE THREAD

**Issue by dhakehurst** [equodev/equo-ide#163]
I am depending on the artop libs from [https://www.artop.org/containers/artop-sdk-update-site-4.13]
this repo actually requires credentials to access (would be nice if this was somehow specifiable...but for now we take a local copy so not an immediate issue).

So I can specify
```
p2deps {
    into("api") {
        //artop
        //p2repo("https://www.artop.org/containers/artop-sdk-update-site-4.13")
        p2repo("<local copy of artop p2 repo>")
        install("org.artop.aal.autosar448")
        install("org.artop.aal.autosar447")
        install("org.artop.aal.autosar40.converters")
        install("org.artop.aal.autosar40.services")
    }
}
```

these IUs depend on a whole selection more that come from repo http://download.eclipse.org/...
i.e.
   p2repo("http://download.eclipse.org/releases/2020-03/202003181000/")
   p2repo("http://download.eclipse.org/sphinx/releases/0.11.x/")

however I am getting errors such as 

```
   > Could not find org.eclipse.emf:org.eclipse.emf.transaction:1.9.1.
     Searched in the following locations:
       - https://repo.maven.apache.org/maven2/org/eclipse/emf/org.eclipse.emf.transaction/1.9.1/org.eclipse.emf.transaction-1.9.1.pom
     Required by:
         project :artop-libs
   > Could not find org.eclipse.emf:org.eclipse.emf.validation:1.8.0.
     Searched in the following locations:
       - https://repo.maven.apache.org/maven2/org/eclipse/emf/org.eclipse.emf.validation/1.8.0/org.eclipse.emf.validation-1.8.0.pom
     Required by:
         project :artop-libs
   > Could not find org.eclipse.emf:org.eclipse.emf.workspace:1.5.1.
     Searched in the following locations:
       - https://repo.maven.apache.org/maven2/org/eclipse/emf/org.eclipse.emf.workspace/1.5.1/org.eclipse.emf.workspace-1.5.1.pom
     Required by:
         project :artop-libs
```

what is causing a dependency on "org.eclipse.emf:org.eclipse.emf.transaction:1.9.1" ??
nothing in my build is specifying this dependency..it looks like a maven/gradlde dependency.

how do I find out what is causing the dependency ?

**Comment by dhakehurst** [comment]
Also, it looks like you rely on mavenCentral being configured as a repository.
I tried to exclude the emf artefacts on mavenCentral:
```
        mavenCentral{
            content {
               excludeGroupByRegex("org\\.eclipse.emf.*")
            }
        }
```

but this causes many more emf dpes (e.g. org.eclipse.emf:org.eclipse.emf.common) not to be found,
and I am sure these are available in a configured p2repo

**Comment by nedtwigg** [comment]
Yeah, you are correct that the problem is EMF dependencies which are being incorrectly translated to MavenCentral coordinates.

https://github.com/equodev/equo-ide/blob/99a7150bc32f6cc8e31d1d12dfc4241eb8151972/solstice/src/main/java/dev/equo/solstice/p2/MavenCentralMapping.java#L58-L64

We could add something like `System.getProperty("equo_forbid_maven)` which could be a space-delimited list of tokens, and if a bundleif contains any of them, then don't do the maven translation. E.g. `System.setProperty('equo_forbid_maven', '.emf')`

If that would work for you, we'd be happy to merge a PR for that.

**Comment by dhakehurst** [comment]
I was thinking something like a list of globs that should be resolved using the p2 repos, i.e.

```
p2deps {
   into(listOf("api")){
     p2repo("...")
     install("...")
     resolutionStrategy {
        useP2(listOf(
          "<glob1>",
          "<glob2>",
          "org.eclipse.emf.**"
         ))
        useMaven(istOf(
          "<glob1>",
          "<glob2>"
         ))
     }
   }
}
```


not sure I have time to implement it at present though ;-)
short term I was thinking either 
a) a simple 'useMaven' Boolean that stops all resolution via maven
b) a check that the dependency actually exists on MavenCentral as part of 'RepoStatus.getJarsOnMavenCentral' - however I'm uncertain how to implement the check.

**Comment by nedtwigg** [comment]
> check that the dependency actually exists on MavenCentral

I think Gradle's dependency model will make this difficult/impossible.

> a simple 'useMaven' Boolean that stops all resolution via maven

Thats seems like a great idea, happy to merge a PR for thar.

**Comment by dhakehurst** [comment]
here is PR, https://github.com/equodev/equo-ide/pull/164

but has some failing tests...let me know how to fix

**Comment by dhakehurst** [comment]
hi, thanks, yes
I think this issue is resolved.
(There is another..but I'll create a different issue for that ;-) )

**Comment by steve-hickman-epistimis** [comment]
It looks like this issues has been resolved by the merged PR (#164). Is there a reason this issue isn't closed?
